email.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+# If running locally (i.e. not on CI), fetch the email from the users git config
+if [ $CI != 'true' ]
+then
+    echo "CI deploy not detected. Using local variables instead"
+    # Fetch the variables from the git config.
+    AUTHOR_EMAIL=$(git config user.email)
+    REACT_APP_DEPLOYS_EMAIL=$REACT_APP_DEPLOYS_EMAIL
+    TASK_NAME=$(git rev-parse HEAD)
+fi
+
+# Validate necessary variables are set
+if [ '$REACT_APP_DEPLOYS_EMAIL' == '' ]
+then
+    echo "REACT_APP_DEPLOYS_EMAIL is not set"
+    exit 1
+fi
+
+if [ '$AUTHOR_EMAIL' == '' ]
+then
+    echo "AUTHOR_EMAIL is not set"
+    exit 1
+fi
+
+if [ '$TASK_NAME' == '' ]
+then
+    echo "Was unable to fetch the task_name or git commit associated with this build"
+    exit 1
+fi
+
+# Fetch previous release tag and get the commits since that tag
+PREVIOUS_VERSION=$(git describe --tags  --abbrev=0  `git rev-list --tags --max-count=1 --skip=1`)
+git log --no-merges $PREVIOUS_VERSION..HEAD --pretty="format:%s (%h)" > body.txt
+
+# Determine which verson of evergreen is available and use that
+if ! [ -x "$(command -v evergreen)" ]
+then
+  echo 'Error: evergreen is not on $PATH.'
+  echo 'Trying to use the local binary'
+  if ! [ -x "$(command -v ~/evergreen)" ]
+  then
+    echo 'Error: Could not find evergreen binary on home directory'
+    exit 1
+  else
+    echo 'Found evergreen binary on home directory'
+    EVERGREEN=~/evergreen
+    CREDENTIALS='-c .evergreen.yml'
```

Comment:
Where did this file come from? I would be surprised if it's already on the host with creds, but maybe I'm forgetting something.

Reply:
This gets generated here. https://github.com/evergreen-ci/spruce/blob/73ece6563534d3fcb10f9193f307ac060fb360a5/.evergreen.yml#L257-L273